Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabFocusHandler.cpp

```diff
@@ -359,56 +359,57 @@ namespace AzToolsFramework::Prefab
+            return InvalidLinkId;
         }
 
         // Grab the owning instance.
-        InstanceOptionalReference owningInstance = m_instanceEntityMapperInterface->FindOwningInstance(entityId);
-        AZ_Assert(owningInstance != AZStd::nullopt, "Owning Instance is null");
+        InstanceOptionalConstReference owningInstance = m_instanceEntityMapperInterface->FindOwningInstance(entityId);
+        AZ_Assert(owningInstance != AZStd::nullopt, "PrefabFocusHandler::AppendPathFromFocusedInstanceToPatchPaths - "
+            "The owning instance of the given entity id is null.");
 
-        // Retrieve the path from the focused prefab instance to the owningInstance of the entityId.
-        AZStd::string prefix;
-        InstanceOptionalReference focusedPrefabInstance = GetInstanceReference(m_rootAliasFocusPath);
+        // Retrieve the path from the focused prefab instance to the owningInstance of the given entity id.
+        InstanceOptionalConstReference focusedInstance = GetInstanceReference(m_rootAliasFocusPath);
+        
+        // Climb up the instance hierarchy from the owning instance until it hits the focused prefab instance.
+        InstanceClimbUpResult climbUpResult = PrefabInstanceUtils::ClimbUpToTargetOrRootInstance(
+            owningInstance->get(), focusedInstance->get());
+        if (!PrefabInstanceUtils::CompareInstances(climbUpResult.reachedInstance, focusedInstance))
+        {
+            AZ_Error("Prefab", false, "PrefabFocusHandler::AppendPathFromFocusedInstanceToPatchPaths - "
+                "Entity id is not owned by a descendant of the focused prefab instance.");
+            return InvalidLinkId;
+        }
 
+        if (!climbUpResult.climbedInstances.empty())
         {
-            // Climb up the instance hierarchy from this instance until you hit the focused prefab instance.
-            InstanceOptionalConstReference instance = owningInstance;
-            AZStd::vector<InstanceOptionalConstReference> instancePath;
+            AZStd::string prefix = "";
 
-            auto climbUpResult = PrefabInstanceUtils::ClimbUpToTargetOrRootInstance(&instance->get(), &focusedPrefabInstance->get());
-            if (climbUpResult.first != &focusedPrefabInstance->get())
+            auto startInstanceIter = ++climbUpResult.climbedInstances.rbegin();
+            for (auto instanceIter = startInstanceIter; instanceIter != climbUpResult.climbedInstances.rend(); ++instanceIter)
             {
-                AZ_Error(
-                    "Prefab", false,
-                    "AppendPathFromFocusedInstanceToPatchPaths - entityId is not owned by a descendant of the focused prefab instance.");
-                return linkId;
+                prefix.append(PrefabDomUtils::PathStartingWithInstances);
+                const Instance& instance = (*instanceIter)->get();
+                prefix.append(instance.GetInstanceAlias());
             }
```

Comment:
Use a local variable to capture `++climbedInstance.rbegin()` to make it more noticeable.

Was trying to use `PrefabInstanceUtils::GetRelativePath()` here but it caused several auto test failed because the helper function does not skip the first instance.